### PR TITLE
Use app api

### PR DESCRIPTION
**Pull request by persandstrom**
This PR is an experiment to move python_verisure from the web api to the app api.

### CODE REVIEW

**Comment by persandstrom** [comment]
@lindehoff Could you take a look this to see if it works as expected? If you have time please do a code review as well.

**Comment by lindehoff** [comment]
Absolutely, I will have a look at it when I get home from a trip to Stockholm.

**Comment by tcastberg** [comment]
@persandstrom How far are we from a stable release of this do you think? Are you noticing any improvements on response times?

**Comment by persandstrom** [comment]
Pretty close @tcastberg. The one major issue is that I'm not able to add support for smartcam since I don't have access to one. @turbokongen, do you have time help out again?

The response time might be a bit better,  but I have not done any measurement.

**Comment by turbokongen** [comment]
I don't have too much time at the moment, but what is your concerns? @persandstrom 
I di not know you have switched to the app api, which is actually great!

**Comment by persandstrom** [comment]
@turbokongen the difference between the web and the app api are too big to guess how the calls are structured. I would rather see that no functionality is lost when switching. I might be able to figure it out with a recording of the network traffic.

**Comment by turbokongen** [comment]
What i get about the smartcam with the curl overview command:
```
{"deviceLabel":"2AGZ DEUT","deviceArea":"Stue","deviceType":"SMARTCAMERA1","temperature":23.9,"time":"2016-12-22T18:11:37.000Z"}],
....
,"smartCameras":[{"motionDetectorActive":false,"area":"Stue","latestImageCapture":"2016-12-16T18:57:49.000Z","deviceLabel":"2AGZ DEUT","numberOfNotViewedImageSeries":1}
....
"customerImageCameras":[{"motionDetectorMode":"INACTIVE","area":"Stue","latestImageCapture":"2016-12-16T18:57:49.000Z","deviceLabel":"2AGZ DEUT","numberOfNotViewedImageSeries":1,"imageResolution":"HIGHER","accelerometerMode":"ACTIVE","capability":"USER_MONITORED_CUSTOMER_IMAGE_CAMERA"}]

**Comment by turbokongen** [comment]
How do you capture the traffic from the app?

**Comment by lindehoff** [comment]
@turbokongen, you'll have to use a web proxy to intersect the traffic. I use https://www.charlesproxy.com and it works great for this.

**Comment by tcastberg** [comment]
Are you imitating the headers of the app to make sure verisure can't easily differentiate calls from the app and calls from python-verisure?

**Comment by persandstrom** [comment]
@tcastberg That might be a good idea, I will go though all the calls and verify that the headers are similar.

**Comment by turbokongen** [comment]
Hmmm... I might be doing this the wrong way...
I downloade Memu(android emulator for windows) and installed the verisure app inside it. Started charles, but no traffic is displayed when using verisure inside memu.
How do you guys do this?

**Comment by persandstrom** [comment]
You can set a proxy on the wireless connection on your physical phone and point it to the IP of the computer running charles. You have to enable ssl ploxying in charles as well for the locations ```e-api02.verisure.com```, ```mymobile.verisure.com```, ```e-api01.verisure.com``` and ```sstats.securitas-direct.com```

**Comment by turbokongen** [comment]
Ok, the format for capture is:
`POST /xbn/2/installation/{{giid}}/device/{{device_id}}/customerimagecamera/imagecapture HTTP/1.1`
And the way of getting all the images is:
`GET /xbn/2/installation/{{giig}}/device/customerimagecamera/imageseries/search?numberOfImageSeries=10&offset=0&fromDate=&toDate=&onlyNotViewed=&_={{token}} HTTP/1.1`
This will return someting like this:
```
{
	"totalNumberOfImageSeries": 2,
	"imageSeries": [{
		"deviceLabel": "2AGZ DEUT",
		"area": "Stue",
		"seriesId": "217437517",
		"viewed": true,
		"image": [{
			"imageId": "7446730110179916413",
			"indexInSeries": "0",
			"captureTime": "2016-12-23T21:19:00.000Z",
			"imageStatus": "COMPLETED",
			"imageOrientation": "NORMAL"
		}]
	}, {
		"deviceLabel": "2AGZ DEUT",
		"area": "Stue",
		"seriesId": "11218876",
		"viewed": true,
		"image": [{
			"imageId": "15228948871282695040",
			"indexInSeries": "0",
			"captureTime": "2016-12-16T18:57:49.000Z",
			"imageStatus": "COMPLETED",
			"imageOrientation": "NORMAL"
		}]
	}]
}
```

**Comment by persandstrom** [comment]
Thank you @turbokongen. I will take a look at it in a few days. Merry Christmas ☺

**Comment by turbokongen** [comment]
I forgot about the last, donload image:
`GET /xbn/2/installation/{{giid}}/device/{{device_id}}/customerimagecamera/image/{{image_id}}/ HTTP/1.1`

**Comment by persandstrom** [comment]
@turbokongen I have added support for cameras now, although everything is untested. If you could do some tests that would be great

**Comment by turbokongen** [comment]
Ok here are the tests:
`imageseries` works like it should.

`capture` gives the following:
```Traceback (most recent call last):
  File "verisure.py", line 3, in <module>
    __main__.main()
  File "/srv/hass/src/python-verisure/verisure/__main__.py", line 242, in main
    print_result(verisure.capture_image(args.device_label), 0)
  File "/srv/hass/src/python-verisure/verisure/__main__.py", line 33, in print_result
    for key, value in overview.__dict__.items():
AttributeError: 'NoneType' object has no attribute '__dict__'
```
 But the command is executed at verisure, and a new image is captured.

`getimage` gives the following:
```
Traceback (most recent call last):
  File "verisure.py", line 3, in <module>
    __main__.main()
  File "/srv/hass/src/python-verisure/verisure/__main__.py", line 250, in main
    args.file_name), 0)
  File "/srv/hass/src/python-verisure/verisure/session.py", line 440, in download_image
    _validate_response(response)
  File "/srv/hass/src/python-verisure/verisure/session.py", line 46, in _validate_response
    response.text.encode('utf-8')))
verisure.session.ResponseError: Invalid response, status code: 400 - Data: b'<response><status><errorGroup>BAD_REQUEST</errorGroup><errorCode>VAL_00028</errorCode><errorMessage>Invalid or non-existing value for device label</errorMessage></status></response>'
``` 
Nothing happens after this.

**Comment by persandstrom** [comment]
Ok thanks, ```capture``` should execute without exception now and ```getimage``` might get further at least...

**Comment by turbokongen** [comment]
Perfect! Now also capture works.
Only `getimage` has some problems.
Replace the double {{ }} in the DOWNLOAD_IMAGE_URL with singles.
After that, the result is the same as capture:
```
Traceback (most recent call last):
  File "verisure.py", line 3, in <module>
    __main__.main()
  File "/srv/hass/src/python-verisure/verisure/__main__.py", line 250, in main
    args.file_name), 0)
  File "/srv/hass/src/python-verisure/verisure/__main__.py", line 33, in print_result
    for key, value in overview.__dict__.items():
AttributeError: 'NoneType' object has no attribute '__dict__'
```
And picture is downloaded.

File: verisure/__main__.py [hunk 1]
```diff
@@ -169,36 +183,72 @@ def main():
...
+            print_result(session.get_climate(args.device_label), 0)
+    if args.command == COMMAND_EVENTLOG:
+        print_result(
+            session.get_history(args.pagesize, args.offset, *args.filter), 0)
+    if args.command == COMMAND_CAPTURE:
+        session.capture_image(args.device_label)
+    if args.command == COMMAND_IMAGESERIES:
+        print_result(session.get_camera_imageseries(), 0)
+    if args.command == COMMAND_GETIMAGE:
+        print_result(
```
Comment (turbokongen) [inline verisure/__main__.py]: Now it fail like the capture command did before fix:
Remove `print_result` and it works flawlessly :+1:

**Pull request merged**